### DISCUSSION TITLE
How to use one modal in multiple nested components?

### DISCUSSION THREAD

**Discussion by andreskrip** [livewire/livewire#4507] [1. Help] [answered]
I have a complex table structure, with many nested components called inside loops.

Entities:
Service - main entity
Partner - can be related with many services
Landing - can be related with many services and partners

![Снимок экрана от 2022-01-11 15-42-25](https://user-images.githubusercontent.com/52978999/148948784-aac3ea6e-4444-49a7-986e-6d0b13b70be5.png)

In general , the logic looks like this:
```
{{--start services component--}}
@foreach ($services as $service)
    ...
    @livewire('monitoring.services.partners', ['serviceId' => $service->id], 's-' . $service->id)
        {{--start partners component--}}
        @foreach ($servicePartners as $partner)
            ...
            @livewire('monitoring.services.partners.landings', ['serviceId' => $service->id, 'partnerId' => $partner->id], 's-' . $service->id . 'p-' . $partner->id)
                {{--start landings component--}}
                @foreach ($partnerLandings as $landing)
                    ...
                    <button wire:click="editLandingStatus({{ $landingId }})" data-toggle="modal" data-target="#editLandingStatus">
                        {{ $landing->name }}
                    </button>
                    ...
                @endforeach
                @once
                    @push('beforeFooter')
                        @include('monitoring.services.partners.landings.edit-landing-status') // modal
                    @endpush
                @endonce
                {{--end landings component--}}
        @endforeach
        {{--end partners component--}}
@endforeach
{{--end services component--}}
```

I don't want to render a large number of same modals (about 40), so I decided to take one up to the main template, but from there the data is not passed to the modal. 

Help me, please, how can I do this?

P.S.:
I tried to load nested components by click, but I didn't find a way to initialize the properties-flags in runtime, because for each set of service(-partner(-landing)) need a unique flag

**Comment by joshhanley** [comment]
@andreskrip you should be able to access the current component inside your modal `@include` using `$this` and then accessing whatever data you need on the component through that. But the issue will be that using `@push` to move the modal, will actually mean that the modal html will be rendered outside the Livewire components root element, as such anything Livewire related then won't work in there.

I wrote a blog article on something similar related to Livewire buttons in headers, check it out, it might give you some more insight about why `@push` could be causing issues https://joshhanley.com.au/articles/how-to-structure-your-layout-file-for-livewire

Hope this helps!

**Comment by andreskrip** [comment, marked as answer]
@joshhanley thanks a lot!
my solution was to use a modal as a livewire component and communicate between the nested component through events with a unique name

**Reply by joshhanley** [reply]
@andreskrip no worries 🙂 ah yep nice!
